walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,60 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.Collections;
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
+  /* Finds all time ranges that satisfy the request given the already existing set of events. */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Find potential conflicts between scheduled events and request.
+    List<TimeRange> potentialConflicts = getPotentialConflicts(events, request);
+
+    // Sort the potential conflicts in order of ascending start time.
+    Collections.sort(potentialConflicts, TimeRange.ORDER_BY_START);
+   
+    Collection<TimeRange> timeRanges = new ArrayList<TimeRange>();
+    int lastConflictEnd = TimeRange.START_OF_DAY;
+    long requestDuration = request.getDuration();
+    for (TimeRange currentConflict : potentialConflicts) {
+        // If the time between the end time of the current conflict and the latest end time 
+        // conflict seen so far is greater that the request duration provided,
+        // add a new time range to the time ranges returned.
+        int currentConflictStart = currentConflict.start();
+        if (currentConflictStart - lastConflictEnd >= requestDuration) {
+            timeRanges.add(TimeRange.fromStartEnd(lastConflictEnd, currentConflictStart, false));
```

Comment:
can add what 'false' is eg`  timeRanges.add(TimeRange.fromStartEnd(lastConflictEnd, currentConflictStart, *exclusive interval */ false));`

Reply:
Done. Good idea - thanks.